src/main/java/org/antlr/intellij/plugin/validation/AddTokenDefinitionFix.java

```diff
@@ -0,0 +1,54 @@
+package org.antlr.intellij.plugin.validation;
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import org.antlr.intellij.plugin.psi.LexerRuleRefNode;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+class AddTokenDefinitionFix extends LocalQuickFixAndIntentionActionOnPsiElement {
+
+    private final LexerRuleRefNode element;
+
+    AddTokenDefinitionFix(LexerRuleRefNode element) {
+        super(element);
+        this.element = Objects.requireNonNull(element);
+    }
+
+    @Nls(capitalization = Nls.Capitalization.Sentence)
+    @NotNull
+    @Override
+    public String getFamilyName() {
+        return "ANTLR4";
+    }
+
+    @NotNull
+    @Override
+    public String getText() {
+        return "Add token definition.";
+    }
+
+    @Override
+    public void invoke(@NotNull Project project, @NotNull PsiFile file, @Nullable("is null when called from inspection") Editor editor, @NotNull PsiElement startElement, @NotNull PsiElement endElement) {
+        appendTokenDefAtLastLine(editor);
+    }
+
+    private void appendTokenDefAtLastLine(@Nullable("is null when called from inspection") Editor editor) {
+        String tokenName = element.getReference().getCanonicalText();
+        int lastLineOffset = editor.getDocument().getLineEndOffset(editor.getDocument().getLineCount() - 1);
+        editor.getDocument().insertString(lastLineOffset, "\n" + buildTokenDefinitionText(tokenName));
+    }
+
+    @NotNull
+    static String buildTokenDefinitionText(String tokenName) {
+        String tokenDefRule = tokenName.toUpperCase().chars().mapToObj(c -> (char) c).map(String::valueOf).collect(Collectors.joining(" "));
```

Comment:
I'm not convinced this is a sensible default value. What about `MYTOKEN: 'MYTOKEN'` with the value automatically selected in the editor to make it easy to modify it?

Reply:
I can add such default as well. I've chosen `MYTOKEN: M Y T O K E N;` because it allows us to create token based on fragments (instead of literal) which is I think better when it comes to case-insensitive languages.

Reply:
OK if you think it's better then go for it. As long as there is the automatic selection, I'm happy :)

Reply:
I'm having problem with automatic selection. I've managed to create selection such as rename selection (is when you press enter the name is accepted). Do you happen to know how to implement this correctly? 

My selection:
![my_token_select](https://user-images.githubusercontent.com/5840392/60702638-c1fdde00-9eff-11e9-993e-2d90ab449ffd.png)

Selection that I want to implement:
![rename_select](https://user-images.githubusercontent.com/5840392/60702650-c7f3bf00-9eff-11e9-85f6-d88766f50d87.png)

Reply:
This could be done using an inline refactoring, but in our case `M Y T O K E N` is not a a `PsiNamedElement` so that won't work. I managed to do a similar thing in Ceylon IDE for IntelliJ [using `TemplateManagerImpl`](https://github.com/eclipse/ceylon-ide-intellij/blob/1f228e03a3eac5643c0928c90613c9203cd28c00/source/org/eclipse/ceylon/ide/intellij/platform/IdeaLinkedMode.ceylon#L95-L106).

Basically you need to:
* create a `TemplateBuilderImpl` on your `PsiFile`
* call `builder.replaceRange(range, new TextExpression("M Y L O O K U P"))` for each text range you want to edit
* call `TemplateManager.getInstance(project).startTemplate(editor, template)` to start the template

By doing this you will basically transform the whole file into a live template with a single editable variable :)

Reply:
Done. I managed to do this without making whole file a template - looked up on ceylon repo :)